### ISSUE TITLE
Remove `smart_auth` / Make an UI widget only

### ISSUE THREAD

**Issue by AlexV525** [Tkko/Flutter_Pinput#180]
The package currently relies on `smart_auth` for automatic verifications. Some users might only want the interface for their implementations like local PIN inputs, which do not use receivers or listeners.

**Comment by praween-link** [comment]
<img width="866" alt="Screenshot 2024-04-09 at 1 13 10 AM" src="https://github.com/Tkko/Flutter_Pinput/assets/67202568/0ddc164e-7aaf-4508-9c58-0c4978ce8ee5">

Same issues

**Comment by Tkko** [comment]
Good idea but  how would you remove smart_auth from Pinput without messing up developer's day :D?

**Comment by AlexV525** [comment]
> Good idea but how would you remove smart_auth from Pinput without messing up developer's day :D?

Probably go with a new package `flutter_pinput`. It's not occupied yet. :)

**Comment by Tkko** [comment]
@AlexV525 I'm thinking of something like this:

I'll create a interface `SmsRetriever` and add it as a `Pinput` property
```dart
abstract class SmsRetriever {
  bool get listenForMultipleSms;

  Future<String?> getSmsCode();

  Future<void> removeSmsListener();
}

class Pinput extends StatefulWidget {
  const Pinput({
    this.smsRetriever,
    /// ...other params
    });
  }
```

In the main app, developers have to implement the `SmsRetriever`, which can use `smart_auth` or any other library for reading sms code

```dart
class SmsRetrieverImpl implements SmsRetriever {
  const SmsRetrieverImpl(this.smartAuth);

  final SmartAuth smartAuth;

  @override
  Future<void> removeSmsListener() {
    return smartAuth.removeSmsListener();
  }

  @override
  Future<String?> getSmsCode() async {
    final res = await smartAuth.getSmsCode();
    if (res.succeed && res.codeFound) {
      return res.code!;
    }
    return null;
  }

  @override
  bool get listenForMultipleSms => false;
}
```
And finally, you can create a Pinput widget:

```dart
    Pinput(
      smsRetriever: SmsRetrieverImpl(
        SmartAuth(),
      ),
    );
```

WDYT?

**Comment by AlexV525** [comment]
> WDYT?

I haven't looked into the implementation, but the idea looks great overall. Some bonus points might be using the `interface` class modifiers if you've already limited the lower bound of the Dart SDK. And thanks for sharing your thoughts!

**Comment by mwendwa-will** [comment]
The idea is great wanted to try it out

**Comment by aitsuki** [comment]
Due to regulatory review requirements, we cannot integrate additional SDKs within our app...

**Comment by AlexV525** [comment]
This seems to have been implemented in v5.0.0. Cheers!

https://github.com/Tkko/Flutter_PinPut/blob/master/MIGRATION.md#migration-to-500
